### PR TITLE
Introduce ElizaMessage.commentIndexOnKeep

### PR DESCRIPTION
**Pull request by ryanpbrewster**
and use it to make Slack pushes for comments fancier

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/discussion/Message.scala [hunk 1]
```diff
@@ -42,6 +42,7 @@ case class CrossServiceMessage(
   isDeleted: Boolean,
   seq: SequenceNumber[Message],
   keep: Id[Keep],
+  commentIndexOnKeep: Option[Int],
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/discussion/Message.scala:45]: `= None` if the message is not a human-generated comment

File: kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageRepo.scala [hunk 2]
```diff
@@ -217,6 +219,15 @@ class MessageRepoImpl @Inject() (
     activeRows.filter(_.keepId === keepId).list
   }
 
+  def getCommentIndex(message: ElizaMessage)(implicit session: RSession): Option[Int] = {
+    if (!message.isActive || message.from.isSystem) None
+    else Some(rows.filter(
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageRepo.scala:224]: should probably be `activeRows` 😞

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackPushGenerator.scala [hunk 3]
```diff
@@ -219,7 +219,7 @@ class SlackPushGenerator @Inject() (
         "  ",
         "View" --> keepLink("viewArticle"),
         "|",
-        "Reply" --> keepLink("reply")
+        ("Reply" + msg.commentIndexOnKeep.filter(_ > 0).map(idx => " $idx").getOrElse("")) --> keepLink("reply")
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackPushGenerator.scala:222]: probably needs a `s"`

File: kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageRepo.scala [hunk 4]
```diff
@@ -217,6 +219,15 @@ class MessageRepoImpl @Inject() (
...
+      m => m.keepId === message.keepId &&
+        (m.from.isDefined || m.nonUserSender.isDefined) &&
+        (m.createdAt < message.createdAt || m.createdAt === message.createdAt && m.id < message.id.get)
+    ).length.run)
+  }
```
Comment (camhashemi) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/model/MessageRepo.scala:229]: nice

**Pull request merged**